press/models.py

```diff
@@ -25,3 +25,61 @@
     'Resource',
     ('data filename media_type sha1'),
 )
+
+
+class CollectionElement:
+    """Represents a collxml element parsed from a Collection XML file.
+    It is iterable and it is comparable using `is_equal_to`.
+    """
+    def __init__(self, element_name, attrs):
+        self.name = element_name
+        self.__children = []
+        self.parent = self
+        self.text = ''
+        # programatically set attributes from data passed-in as a hash
+        for k, v in attrs.items():
+            uri, localname = k
+            setattr(self, localname, v)
+
+    def __repr__(self):
+        return 'Element({})'.format(self.name)
+
+    def add_child(self, child):
+        """Works like append for XML ElementTree-s.
+        """
+        if isinstance(child, self.__class__):
+            child.parent = self
+            self.__children.append(child)
+            return self
+        else:
+            raise Exception("must be of the same type.")
+
+    def __iter__(self):
+        return iter(self.__children)
+
+    def __len__(self):
+        return len(self.__children)
+
+    def traverse(self):
+        yield self
+
+        for nested in self:
+            yield from nested.traverse()
+
+    def getparent(self):
+        return self.parent
+
+    def is_equal_to(self, other):
+        """Equality is defined as two collections having the same
+        (type of) elements in the same order and all modules having
+        the same title.
+        """
+        if self.name == 'title' and other.name == 'title':
+            # title tags may have nested tags within them,
+            # so consider the text in those as well.
+            return self._complete_title() == other._complete_title()
+        else:
+            return self.name == other.name
```

Comment:
Lookup special method `__eq__`, so that `tree == tree` works.

Reply:
I had implemented that method, actually, but I received very strong pushback against it from @kevinburleigh75—basically because it may cause more trouble in the future than it's worth, and @reedstrm seemed to agree. I considered, however, implementing it in a separate object/data structure whose sole purpose is to be used for comparing *and* diffying two trees, but I didn't get to that point yet, not in this PR at least. That specialized tree structure would also implement `__hash__` in addition to `__eq__` (https://docs.python.org/3.7/reference/datamodel.html#object.__hash__).